### PR TITLE
Fixing the issues in the tests

### PR DESCRIPTION
**Pull request by kmp5VT**
# Description

This PR seeks to fix the issues pointed out by Matt in the initial commit of this repo.

Fixes #(issue)

I added a extensions for FillArrays and created default type parameters for an AbstractFill.

# Checklist:

- [x] Default parameters are defined for AbstractFill and AbstractUnallocatedFill types.
- [x] Tests all run as expected with no errors

### CODE REVIEW

**Comment by kmp5VT** [comment]
@mtfishman It looks like the problem was related to looping over the types written as sybmols. For example in `for typ in (:Fill, :Zeros)`, `typ` was being interpreted as a symbol and, though we had defined a default_type_parameter for the `AbstractFill` type, we don't have default parameters defined for symbols which was causing the code to error out.

File: working_example/Project.toml
Comment (mtfishman) [inline working_example/Project.toml:1]: Looks like this should be deleted.

File: test/test_defaults.jl [hunk 1]
```diff
@@ -0,0 +1,79 @@
+
+## TODO: These are broken, fix them. Also, split off into a
+## seperate test file, and maybe make a package extension?
+@eval module $(gensym())
+using Pkg
+Pkg.add(; url="https://github.com/ITensor/TypeParameterAccessors.jl")
```
[the remark this answers is not in the document]
Comment (mtfishman) [reply test/test_defaults.jl:6]: Also note that once #6 is merged this won't be needed.

[the remark this answers is not in the document]
Comment (kmp5VT) [reply test/test_defaults.jl]: Thanks, I saw that PR as well. I added `TypeParameterAccessors`'s information to the `Project.toml` and the tests run now. I also was able to remove the `Pkg.add` call in the `test_default.jl` file

File: working_example/Project.toml
Comment (kmp5VT) [reply working_example/Project.toml:1]: Deleted, thanks!

**Comment by lkdvos** [comment]
@mtfishman , I think the easiest would be if you merge this, I'll apply the new skeleton on top of that in a separate PR?

**Pull request merged**